subsys/bluetooth/mesh/access.c

```diff
@@ -448,9 +438,9 @@ int bt_mesh_comp_data_get_page_0(struct net_buf_simple *buf, size_t offset)
 	for (i = 0; i < comp->elem_count; i++) {
 		int err;
 
-		err = comp_add_elem(buf, &comp->elem[i], &offset);
+		err = comp_add_elem(buf, &comp->elem[i], &offset, split_elems);
 		if (err) {
-			return err;
+			return 0;
```

Comment:
Because `comp_add_elem` returning ENOBUFS simply means "there is no more room for elements now". This is expected to happen, so then the loop is done and we have successfully added as many elements as we can, and can return success.